Collection Inspection #SCL-9841 convert map.find(_ == k).map(_._2) to map(k)

mattfowler opened the pull request:
Adds the functionality described in #SCL-9841 to convert map.find(_ == k).map(_._2) to map(k). Inspection is named "find and map to apply" but there may be a better name for this inspection.

mattfowler committed:
Adding inspection #SCL-9841 - convert map.find(_ == k).map(_._2) to map(k)

mattfowler committed:
Adding inspection #SCL-9841 - convert map.find(_ == k).map(_._2) to map(k)

mattfowler committed:
Adding inspection #SCL-9841 - add local inspection in common xml.

mattfowler committed:
Adding inspection #SCL-9841 - use consistent language in description.

niktrop commented:
@mattfowler Thank you for contribution!
You need to update name of the description file, it should match inspection short name in the `scala-plugin-common.xml`. And it would be better to have all your changes in a single commit.

mattfowler committed:
Merge remote-tracking branch 'origin/idea16.x' into idea16.x

mattfowler commented:
@niktrop No problem! I updated the name of the description file and squashed the commits into commit b9c22d2. Not sure if any other branch manipulation is needed.  Let me know if there is.
